client/client.go

```diff
@@ -49,11 +50,19 @@ type Client struct {
 	server neoFSAPIServer
 }
 
+var errNonNeoSigner = fmt.Errorf("%w: expected ECDSA_DETERMINISTIC_SHA256 scheme", neofscrypto.ErrIncorrectSigner)
+
 // New creates an instance of Client initialized with the given parameters.
 //
 // See docs of [PrmInit] methods for details. See also [Client.Dial]/[Client.Close].
+//
+// Returned errors:
+//   - [neofscrypto.ErrIncorrectSigner]
 func New(prm PrmInit) (*Client, error) {
 	var c = new(Client)
+	if prm.signer != nil && prm.signer.Scheme() != neofscrypto.ECDSA_DETERMINISTIC_SHA256 {
```

Comment:
in fact, i did not expect that check here, is it mandatory?

Reply:
Any other signer scheme leads to container management failures.

Reply:
yes but `Client` is not meant to work with containers in its every call?

cc @cthulhu-rider, @smallhive

Reply:
As I remember, the `Client` works in general with `ECDSA_DETERMINISTIC_SHA256` signer, not only container operations. I don't mind to have this check here.

Another option is moving this condition to `PrmInit.SetDefaultSigner`, but in this way the requirement to sheme will be not so transparent

Reply:
> yes but Client is not meant to work with containers in its every call?

Sure, just omit the default signer and pass it explicitly to every call.

Reply:
> Sure, just omit the default signer and pass it explicitly to every call.

and what if i have 50 calls in my code and none of it relates to containers? 50 repeated lines when we have `SetDefault`? once again: i just did not expect it to be here but if everyone is ok with that, i can be ok too

Reply:
> I'd even say that ECDSA_SHA512 should not had been created at all

I'd say that it can be dropped then. Since we already have it, I'd support everything as a default but error every method that is not possible without a predefined scheme